conan/api/subapi/cache.py

```diff
@@ -133,9 +131,10 @@ def save(self, package_list, tgz_path, no_source=False):
         mkdir(os.path.dirname(tgz_path))
         name = os.path.basename(tgz_path)
         compresslevel = global_conf.get("core.gzip:compresslevel", check_type=int)
+
         with open(tgz_path, "wb") as tgz_handle:
-            tgz = gzopen_without_timestamps(name, fileobj=tgz_handle,
-                                            compresslevel=compresslevel)
+            tgz = tar_compressor(name, fileobj=tgz_handle, compresslevel=compresslevel,
```

Comment:
This is the place where a previous refactor could make sense (in a separate PR):

- Let the code build a dict/list of the things to add to the tgz, but not really call ``tgz.add()`` in the loops
- Do a single call later with that dict/list of the things to add.

that will leave the code ready for introducing the ``if plugin -> use plugin`` idiom without having to move things around